canonical/maas-ui
# fix: kvm page non-admin error

## Done

- kvm page uncaught error when viewing as a non-admin user (when power_parameters are undefined

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Log in as a non-admin user (credentials in lastpass) to the demo
- Go to the KVM page `/MAAS/r/kvm/virsh/5/resources` (you can compare it with http://polong.internal:5240/MAAS/r/kvm/virsh/5/resources where the issue occurs)
- You should not see the "An unexpected error has occurred, please try refreshing your browser window." error.

## Fixes

Fixes: https://github.com/canonical/maas-ui/issues/4442

## Launchpad issue

Related Launchpad maas issue in the form `lp#number`.

## Backports

In general, please propose fixes against _main_ rather than release branches (e.g. 2.7), unless the fix is only applicable for that specific release. Please apply backport labels to the PR (e.g. "Backport 2.7") for the appropriate releases to target.

Only bug and security fixes should be backported, new features should only land in main.
